pylint/checkers/base/basic_checker.py

```diff
@@ -317,6 +317,17 @@ def _check_using_constant_test(
         emit = isinstance(test, (nodes.Const,) + structs + const_nodes)
         if not isinstance(test, except_nodes):
             inferred = utils.safe_infer(test)
+        # Emit if the call is to a function that only returns GeneratorExp (always truthy)
+        elif isinstance(test, nodes.Call):
+            inferred_call = utils.safe_infer(test.func)
+            if isinstance(inferred_call, nodes.FunctionDef):
+                return_nodes = list(inferred_call._get_return_nodes_skip_functions())
+                if return_nodes and all(
+                    isinstance(n.value, nodes.GeneratorExp) for n in return_nodes
+                ):
```

Comment:
```suggestion
                if all(
                    isinstance(n.value, nodes.GeneratorExp)
                    for n in inferred_call._get_return_nodes_skip_functions()
                ):
```
I guess you had a reason to generate the full list each time (it can be None instead of being simply empty ?), but wouldn't it be faster like this because we can shortcut when one of the value is not a generator expression ?

Reply:
Yah I wanted to avoid two loops, but I need to return false if empty, and `all` returns True.

I could create some booleans and track them during a loop.

Reply:
I always waffle on what's the clearest way to do this pattern.

Reply:
I'm a little lost there. I'm under the impression that the design of ``_get_return_nodes_skip_functions`` prevents the use of a generator because we have to do a special case for when it's empty, is that right ?

Reply:
My bad for not taking the time to explain fully. `_get_return_nodes_skip_functions` returns a generator. I just can't do `all()` because that returns True for empty iterables, and I don't want to emit a message if the function doesn't have return statements.